test-apps/display-test-app/ios/imodeljs-test-app/imodeljs-test-app/DtaServiceAuthorizationClient.swift

```diff
@@ -0,0 +1,129 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import Foundation
+
+/// A struct to hold the settings used by DtaOidcAuthorizationClient
+struct DtaServiceAuthSettings {
+    var clientId: String
+    var clientSecret: String
+    var scope: String
+    var authority: String
+}
+
+typealias DtaServiceAuthorizationClientCallback = (Error?) -> ()
+
+class DtaServiceAuthorizationClient: NSObject, DtaAuthorizationClient {
+    /// Instance for `onAccessTokenChanged` property from the `AuthorizationClient` protocol.
+    public var onAccessTokenChanged: AccessTokenChangedCallback?
+    /// Instance for `errorDomain` property from the `DtaAuthorizationClient` protocol.
+    public let errorDomain = "com.bentley.display-test-app"
+    var issuerURL = URL(string: "https://ims.bentley.com")!
+    var authSettings: DtaServiceAuthSettings
+    let logger = PrintLogger()
+    var userInfo: NSDictionary?
+    var accessToken: String?
+    var expirationDate: Date?
+
+    init?(configData: JSON) {
+        let clientId = configData["IMJS_OIDC_CLIENT_ID"] as? String ?? ""
+        let clientSecret = configData["IMJS_OIDC_CLIENT_SECRET"] as? String ?? ""
+        let scope = configData["IMJS_OIDC_SCOPE"] as? String ?? "email openid profile organization itwinjs"
+        let authority = configData["IMJS_SERVICE_AUTHORITY"] as? String ?? "https://ims.bentley.com"
+        authSettings = DtaServiceAuthSettings(clientId: clientId, clientSecret: clientSecret, scope: scope, authority: authority)
+        super.init()
+        if checkSettings() != nil {
+            return nil
+        }
+        var prefix = configData["IMJS_URL_PREFIX"] as? String
+        if let authorityURL = URL(string: authority) {
+            if var authorityURLComponents = URLComponents(url: authorityURL, resolvingAgainstBaseURL: false) {
+                if prefix != nil {
```

Comment:
These 3 ifs can be combined into 1 to improve readability by having only one set of braces. Also, the `prefix != nil` check should be done first.

Reply:
Fixed.